### PR TITLE
latex table option

### LINKED ISSUE #458: Ordered results tables
I know results tables are written from dicts, but if there were a way to print the Variable and Constant tables in alphabetical order, and the Sensitivity table in the order of descending magnitude, I think it would be a really cool enhancement!

It's getting a bit tricky to find the value I'm looking for as my programs get to be a little bigger.

### PR DESCRIPTION
**Pull request by whoburg**
@pgkirsch says:
"This is the code I modified to create latex tables directly from solution arrays. Any feedback is welcome."

Resolved merge conflict.

@pgkirsch, post example output?

@bqpd, what do you think of this?

(later, @bqpd notes that this PR now closes #458)

### CODE REVIEW

**Comment by bqpd** [comment]
Add a screenshot of example output?

**Comment by whoburg** [comment]
retest this please

File: gpkit/solution_array.py [hunk 1]
```diff
@@ -203,17 +209,24 @@ def results_table(data, title, minval=0, printunits=True, fixedcols=True,
...
-        assert len(list(dirs)) == len(list(maxlens))
-        fmts = ['{0:%s%s}' % (direc, L) for direc, L in zip(dirs, maxlens)]
-    lines = [[fmt.format(s) for fmt, s in zip(fmts, line)] for line in lines]
-    lines = [title] + ["-"*len(title)] + [''.join(l) for l in lines] + [""]
+        if latex == False:
+            lines.append([varstr, valstr, units, label])
+        else:
+            varstr = varstr.replace(" : ", "")
+            lines.append(["$", varstr, "$ & ",  valstr, " & ",  units, " & ", label, " \\\\"])
+    if latex == False:
```
Comment (whoburg) [inline gpkit/solution_array.py]: @pgkirsch, this should be `if not latex`.

**Comment by whoburg** [comment]
What's the plan / next step for this PR?

**Comment by bqpd** [comment]
I'd propose adding the sorting and then merging it?

**Comment by bqpd** [comment]
Implemented sorting and cleaned up the `table()` code; ready to merge this after a review.

File: gpkit/solution_array.py [hunk 2]
```diff
@@ -90,62 +97,51 @@ def sens(self, p):
...
+                    costs = ["%-8.3g" % c for c in subdict[:4]]
+                    strs += [" [ %s %s ]" % ("  ".join(costs),
+                                             "..." if len(self) > 4 else "")]
+                    cost_units = self.program[0].cost.units
+                else:
+                    strs += [" %-.4g" % subdict]
+                    cost_units = self.program.cost.units
+                strs[-1] += unitstr(cost_units, into=" [%s] ", dimless="")
+                strs += [""]
+            elif not subdict:
```
Comment (whoburg) [inline gpkit/solution_array.py]: This means the `table` specified cannot be printed, right? Should we print a warning then?

File: gpkit/solution_array.py [hunk 3]
```diff
@@ -90,62 +97,70 @@ def sens(self, p):
...
+        ---------
+        tables: Iterable
+            Which to print of ("cost", "sweepvariables", "freevariables",
+                               "constants", "sensitivities")
+        fixedcols: If true, print vectors in fixed-width format
+        latex: If true, return latex format; otherwise plain text
+        included_models: Iterable
+            If specified, the models to include
+        excluded_models: Iterable
+            If specified, models to exclude
```
Comment (whoburg) [inline gpkit/solution_array.py]: @bqpd, this should be updated to include the `type` of the models in `included_models` and `excluded_models`. Also in the `results_table` docstring

File: gpkit/solution_array.py (hunk 2 above)
Comment (bqpd) [inline gpkit/solution_array.py]: Sounds good!

File: gpkit/solution_array.py [hunk 4]
```diff
@@ -90,62 +97,70 @@ def sens(self, p):
...
-            strs += ["\nCost\n----"]
-            if len(self) > 1:
-                costs = ["%-8.3g" % c for c in self["cost"][:4]]
-                strs += [" [ %s %s ]" % ("  ".join(costs),
-                                         "..." if len(self) > 4 else "")]
-                cost_units = self.program[0].cost.units
+        for table in tables:
+            subdict = self.get(table, None)
+            table_title = self.table_titles[table]
+            if table == "cost":
```
Comment (whoburg) [inline gpkit/solution_array.py]: ah, this is much better, now users can reorder their tables easily

File: gpkit/solution_array.py (hunk 3 above)
Comment (bqpd) [inline gpkit/solution_array.py]: Yup!

**Comment by whoburg** [comment]
This is a great improvement! @bqpd, couple inline comments; I'm otherwise ready to merge.

**Pull request merged**